### ISSUE TITLE
New MaStR Version

### ISSUE THREAD

**Issue by deniztepe** [OpenEnergyPlatform/open-MaStR#348]
A new MaStR Release is due to 01.10.2022. I have read that few columns are going to be edited. I thought we can sum up possible adaptations under this issue. 

- [x] Dropped columns
- [x] New columns 
---
### Release Notes
https://test.marktstammdatenregister.de/MaStRHilfe/subpages/releasenotes.html

### konkrete Änderung in API / ORM
https://test.marktstammdatenregister.de/MaStRHilfe/files/webdienst/WSDL_Patchnotes_V1.2.87.xlsx

**Comment by Wobsley** [comment]
Since the new release on 1 Oct 2022 our API calls no longer work. We noticed in the release notes (link in original post) that there is a reference to a new web service "Bereitstellung der neuen Webdienstversion im Produktions- und Testsystem".  Are other users experiencing issues with the API calls using open_mastr.soap_api.download?

**Comment by Wobsley** [comment]
The website https://www.marktstammdatenregister.de/MaStRHilfe/subpages/webdienst.html provides links to the wsdls.
https://www.marktstammdatenregister.de/MaStRAPI/wsdl/mastr.wsdl points to the development endpoints which do not resolve (see the services endpoint entwicklungs.marktstammdatenregister). But
https://test.marktstammdatenregister.de/MaStRAPI/wsdl/mastr.wsdl correctly points to the live endpoints.
As a result, I have switched to using the test wsdl and have contacted MaStR to confirm whether the two wsdls have simply been misnamed.

**Comment by FlorianK13** [comment]
Hi @Wobsley, thanks for your comments. We'll have a look at it soon.

**Comment by FlorianK13** [comment]
I get the following error when downloading extended unit data for wind using the develop branch:

TypeError: 'Buergerenergie' is an invalid keyword argument for WindExtended

**Comment by chrwm** [comment]
> Since the new release on 1 Oct 2022 our API calls no longer work. We noticed in the release notes (link in original post) that there is a reference to a new web service "Bereitstellung der neuen Webdienstversion im Produktions- und Testsystem". Are other users experiencing issues with the API calls using open_mastr.soap_api.download?

Establishing a connection to the API and downloading works for me (on production & develop branch)

> I get the following error when downloading extended unit data for wind using the develop branch: TypeError: 'Buergerenergie' is an invalid keyword argument for WindExtended

However, due to the current [changes](https://test.marktstammdatenregister.de/MaStRHilfe/files/webdienst/WSDL_Patchnotes_V1.2.87.xlsx) in column naming (orm) for some datapoints, the download breaks at some point, as observed by FlorianK13.
I'll open an issue to implement the specific changes from the WSDL_Patchnote so that this issue will remain more general.

**Comment by chrwm** [comment]
@Wobsley I adapted the code to the wsdl changes in a [feature branch](https://github.com/OpenEnergyPlatform/open-MaStR/tree/feature-351-consider_wsdl_changes) #351 (branched off from develop), which could be merged to production in the coming week. Now, on my end, the standard API calls from the main.py script work again. 

> Since the new release on 1 Oct 2022 our API calls no longer work.  

Are you using the calls from the main.py script or additional calls? Note: I just fixed the ORM database structure for calls from the main.py script.
